### PR TITLE
Add unit tests for JInstallerAdapter

### PR DESCRIPTION
**Pull request by wilsonge**
This gives us code coverage of ~75% of the lines in the class (15/23 methods in the class)

### CODE REVIEW

File: tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php [hunk 1]
```diff
@@ -0,0 +1,1618 @@
...
+	/**
+	 * @testdox Tests the public constructor
+	 */
+	public function testConstructor()
+	{
+		$mockInstaller = $this->getMock('JInstaller');
+		$mockDatabase = $this->getMockDatabase();
+		$object = $this->getMockForAbstractClass('JInstallerAdapter', array($mockInstaller, $mockDatabase, array('foo' => 'bar')));
+
+		$this->assertInstanceOf(
```
Comment (mbabker) [inline tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php]: `$this->assertAttributeInstanceOf('JTableExtension', 'extension', 'object');`

Don't need to use our Reflection helpers here.  Use the PHPUnit API as it's practical.

File: tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php [hunk 2]
```diff
@@ -0,0 +1,1618 @@
...
+			'JTableExtension',
+			TestReflection::getValue($object, 'extension')
+		);
+
+		$this->assertInstanceOf(
```
Comment (mbabker) [inline tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php]: `$this->assertAttributeSame($mockDatabase, 'db', $object);`

If you're injecting the database in, I'd validate that what gets stored is what you injected and not just make sure the property is an instance of the expected class (which may be a different instance than what you're looking for; test specificity).

File: tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php (hunk 1 above)
Comment (wilsonge) [inline tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php]: Fixed and a couple of other instances based on this

File: tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php [hunk 3]
```diff
@@ -0,0 +1,1611 @@
...
+ */
+
+/**
+ * Test class for JInstallerAdapter.
+ *
+ * @package     Joomla.UnitTest
+ * @subpackage  Installer
+ * @since       3.4.4
+ */
+class JInstallerAdapterTest extends TestCaseDatabase
```
Comment (mbabker) [inline tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php]: If you've mocked everything and you aren't running any real database queries (still looking this over), then you can just extend `TestCase` instead.  Saves a few processing cycles on the test setup if it doesn't have to deal with the database configuration.

File: tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php [hunk 4]
```diff
@@ -0,0 +1,1611 @@
...
+			->with(array('element' => $element, 'type' => $type));
+
+		TestReflection::setValue($object, 'extension', $mockTableExtension);
+		TestReflection::setValue($object, 'type', $type);
+		TestReflection::setValue($object, 'element', $element);
+
+		// Invoke the method
+		TestReflection::invoke($object, 'checkExistingExtension');
+
+		$this->assertEquals(
```
Comment (mbabker) [inline tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php]: `$this->assertAttributeEquals($extensionId, 'currentExtensionId', $object, 'The extension ID was not populated correctly for a found extension');

**Comment by mbabker** [comment]
Can you put `@covers` tags on all the test methods please?

**Comment by wilsonge** [comment]
Done everything I think

**Comment by mbabker** [comment]
Well, the question was answered on the database thing.  Guess it needs to extend TestCaseDatabase.

**Comment by wilsonge** [comment]
I'm going to try mocking the ```JFactory::$database``` in the constructor. That might be enough. If not I'll revert it back to the way it was

File: tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php (hunk 3 above)
Comment (wilsonge) [inline tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php]: OK So my dataset still isn't being loaded :(

Comment (mbabker) [inline tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php]: Stop overriding `JFactory::$database` in your setup method.  `TestCaseDatabase` is setting the test connection in its setup routine.

Comment (wilsonge) [inline tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php]: Ahh fair. Does this mean I don't need to inject the extensions data set as well?

Comment (mbabker) [inline tests/unit/suites/libraries/cms/installer/JInstallerAdapterTest.php]: Long and short, just make sure all the changes from https://github.com/wilsonge/joomla-cms/commit/6ab866e1a433f57bd7a416142c0f1c1baf3450fa are reverted and we can go from there.

**Comment by wilsonge** [comment]
@mbabker where we at with this?

**Pull request merged**